src/test/java/io/confluent/idesidecar/restapi/resources/ConnectionsResourceTest.java

```diff
@@ -1349,6 +1437,73 @@ record TestInput(
                 .withSource("ccloud_config")
                 .withDetail("CCloud configuration is not allowed when type is DIRECT")
         ),
+        new TestInput(
+            "Direct spec is invalid with SSL without truststore path",
+            """
+            {
+              "name": "Connection 1",
+              "type": "DIRECT",
+              "kafka_cluster": {
+                "bootstrap_servers": "localhost:9092",
+                "ssl": {
+                  "truststore": {
+                    "password": "truststore-password"
+                  }
+                }
+              }
+            }
+            """,
+            createError()
+                .withSource("kafka_cluster.ssl.truststore.path")
+                .withDetail("Kafka cluster truststore path is required and may not be blank")
+        ),
+        new TestInput(
+            "Direct spec is invalid with SSL having keystore only",
```

Comment:
Can't we rely on the built-in/system certs in the absence of the truststore?
AFAIK, that's how you'd [use mTLS with, for instance, Confluent Cloud](https://docs.confluent.io/cloud/current/security/authenticate/workload-identities/identity-providers/mtls/configure.html#configure-mtls).

Reply:
Sounds good -- I'd imagine the built-in/system certs will be picked up automatically by the Java runtime without us having to configure anything.

Reply:
We should definitely test if that works with the native executable.

Reply:
Sounds good -- I've already created https://github.com/confluentinc/ide-sidecar/issues/235 for testing mTLS against Confluent Cloud, testing would be taken care of as part of that.

Reply:
For now, I've changed the constraints to allow passing keystore options without truststore.